subprojects/gdk_rust/gdk_electrum/src/lib.rs

```diff
@@ -1037,125 +943,142 @@ impl<S: Read + Write> Syncer<S> {
             }
         }
 
-        self.db.insert_index(Index::External, last_used[Index::External as usize])?;
-        self.db.insert_index(Index::Internal, last_used[Index::Internal as usize])?;
-        trace!("last_used: {:?}", last_used,);
-
-        let new_txs = self.sync_txs(&history_txs_id)?;
-        self.sync_headers(&heights_set)?;
-        self.sync_height(&txid_height)?;
-
-        if new_txs {
-            self.db.flush()?;
-        }
-
-        trace!("elapsed {}", start.elapsed().as_millis());
+        let new_txs = self.download_txs(&history_txs_id, &scripts, &client)?;
+        let headers = self.download_headers(&heights_set, &client)?;
+
+        let store_indexes = self.store.read()?.cache.indexes.clone();
+
+        let changed = if !new_txs.txs.is_empty()
+            || !headers.is_empty()
+            || store_indexes != last_used
+            || !scripts.is_empty()
+        {
+            debug!("There are changes in the store");
+            let mut store_write = self.store.write()?;
+            store_write.cache.indexes = last_used;
+            store_write.cache.all_txs.extend(new_txs.txs.into_iter());
+            store_write.cache.unblinded.extend(new_txs.unblinds);
+            store_write.cache.headers.extend(headers);
+            store_write.cache.heights.clear(); // something in the db is not in live list (rbf), removing
```

Comment:
The comment seems to be out of place.

Reply:
I am afraid it was only very unclear, updated in 9434076766079051f03e337d589bbda86b99a8b2